pkg/danm/danm_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+  "testing"
+)
+
+var devicePool0 = "pool0"
+var devicePool1 = "pool1"
+
+var allocatedDevices = make(map[string]*[]string)
+
+func TestPopDevicePanic(t *testing.T) {
```

Comment:
this test case is one of the most common antip-atterns of unit testing

if you find an error within the production code with your unit test, you should correct the error in the code, rather than glorifying it with a test case
in this case popDevice shall return an error to the caller if there is nothing to pop

Reply:
FYI: this is not an FYI comment like the other two, this is malpractice

on the other hand: congrats, you found an error in the code! :)